src/wp-admin/setup-config.php

```diff
@@ -438,10 +438,12 @@ function setup_config_display_header( $body_classes = array() ) {
 		}
 
 		$handle = fopen( $path_to_wp_config, 'w' );
-		foreach ( $config_file as $line ) {
-			fwrite( $handle, $line );
+		if ( false !== $handle ) {
+			foreach ( $config_file as $line ) {
+				fwrite( $handle, $line );
+			}
+			fclose( $handle );
 		}
```

Comment:
Needs to catch if a failure happens to alert that the config is not going to be written, i.e. similar to:

```
			<?php
			/* translators: %s: wp-config.php */
			printf( __( 'Unable to write to %s file.' ), '<code>wp-config.php</code>' );
			?>
```